consensus/hotstuff_leader.go

```diff
@@ -200,6 +201,17 @@ func (handler *HotstuffLeaderMessageHandler) HandleCommitMessage(m *consensusMod
 	// There is no "replica behavior" to imitate here
 
 	m.paceMaker.NewHeight()
+
+	m.
+		GetBus().
+		GetTelemetryModule().
+		IncCounter("consensus_blockchain_height")
+
+	fmt.Println("Incremented blockchain height counter")
+	m.
+		GetBus().
+		GetTelemetryModule().
+		SetGauge("p2p_broadcast_msg_received_total_per_block", 0)
```

Comment:
If we are bouncing gauge to zero on every new block, then we likely never will see a correct number in prometheus/vm/grafana. We are currently gathering the metrics with "pull" model - meaning an agent periodically scrapes the metrics from the exporter endpoint. Currently, the agent scrapes the telemetry every 5 second. So we almost guaranteed will lose the real number of `msg_received_total_per_block` because it is most likely going to be set to 0 in between scrapes. Does that make sense?

The potential solutions I see are:
1. Make this a counter `msg_received`, and do not care about per block part - we could calculate approximate number of messages with somewhat sophisticated query (I am not yet sure how to write, but can look into that) - something like for every increase of `blockchain_height` metric, calculate an increase of `msg_received`.
2. If we need a 100% correct number of msgs per block, not an approximate value, then we can make this a log entry with count per block (e.g. `msg="recieved messages" block=10 msg_rec=300`) when new block introduced. 
3. We look/switch into push model and trigger flush not only periodically, but also when some sort of event happens, such as new block introduction. This would work with VictoriaMetrics, probably will work with Prometheus, but can't tell for sure about some other data stores.

Reply:
Seems like Summary metric type is the way to go: https://prometheus.io/docs/concepts/metric_types/#summary